packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -1140,12 +1183,116 @@ class UiKitViewController {
   /// The [UiKitViewController] object is unusable after calling this.
   /// The `id` of the platform view cannot be reused after the view is
   /// disposed.
+  @override
   Future<void> dispose() async {
     _debugDisposed = true;
     await SystemChannels.platform_views.invokeMethod<void>('dispose', id);
   }
 }
 
+/// Controls an Linux GtkWidget.
+///
+/// Typically created with [PlatformViewsService.initGtkView].
+class GtkViewController extends GestureRecognitionPlatformViewController {
+  GtkViewController._(
+    this.id,
+    TextDirection layoutDirection,
+  ) : assert(id != null),
+      assert(layoutDirection != null),
+      _layoutDirection = layoutDirection;
+
+  /// Gtk's [GTK_TEXT_DIR_None](https://developer.gnome.org/gtk3/stable/GtkWidget.html#GtkTextDirection) value.
+  static const int kGtkTextDirectionNone = 0;
+
+  /// Gtk's [GTK_TEXT_DIR_LTR](https://developer.gnome.org/gtk3/stable/GtkWidget.html#GtkTextDirection) value.
+  static const int kGtkTextDirectionLtr = 1;
+
+  /// Gtk's [GTK_TEXT_DIR_RTL](https://developer.gnome.org/gtk3/stable/GtkWidget.html#GtkTextDirection) value.
+  static const int kGtkTextDirectionRtl = 2;
+
+  /// The unique identifier of the Linux GtkWidget controlled by this controller.
+  @override
+  final int id;
+
+  bool _debugDisposed = false;
+
+  TextDirection _layoutDirection;
+
+  static int _getGtkTextDirection(TextDirection direction) {
+    assert(direction != null);
+    switch (direction) {
+      case TextDirection.ltr:
+        return kGtkTextDirectionLtr;
+      case TextDirection.rtl:
+        return kGtkTextDirectionRtl;
+    }
+  }
+
+  /// Sets the layout direction for the Linux GtkWidget.
+  @override
+  Future<void> setLayoutDirection(TextDirection layoutDirection) async {
+    assert(!_debugDisposed, 'trying to set a layout direction for a disposed Linux GtkWidget. View id: $id');
+
+    if (layoutDirection == _layoutDirection)
+      return;
+
+    assert(layoutDirection != null);
+    _layoutDirection = layoutDirection;
+
+    final List<int> args = <int>[id, _getGtkTextDirection(layoutDirection)];
+    return SystemChannels.platform_views.invokeMethod('setDirection', args);
+  }
+
+  /// Accept an active gesture.
+  ///
+  /// When a touch sequence is happening on the embedded GtkWidget all touch events are delayed.
+  /// Calling this method releases the delayed events to the embedded UIView and makes it consume
+  /// any following touch events for the pointers involved in the active gesture.
+  @override
+  Future<void> acceptGesture(int pointer) {
+    final List<int> args = <int>[id, pointer];
+    return SystemChannels.platform_views.invokeMethod('acceptGesture', args);
+  }
+
+  /// Rejects an active gesture.
+  ///
+  /// When a touch sequence is happening on the embedded GtkWidget all touch events are delayed.
```

Comment:
Same, this only discusses touch.

More generally: is this overall description accurate? Is the Linux platform view input system build on a delay system the way iOS is? Is there a design doc describing the approach?